### ISSUE TITLE
bole harvest litter logic

### ISSUE THREAD

**Issue by rgknox** [NGEET/fates#489]
We have 4 coarse woody debris pools.  The last pool is for the trunk.  In the code that handles distribution of tree mortality into the different litter pools, we have some logic that handles harvesting differently for trunk and non-trunk components:

https://github.com/NGEET/fates/blob/master/biogeochem/EDPhysiologyMod.F90#L1229

However, it looks like the logic is backwards.  When c==ncwd, we should be making provisions for exporting harvested wood, and when c/=ncwd, we should be operating normaly.

@huangmy @ckoven

**Comment by rgknox** [comment]
I think the litter generation is fine, its just the trunk product and the exported carbon should be moved to the first condition.

**Comment by huangmy** [comment]
@rgknox @ckoven 
Just to make sure you capture my comments here. Ryan is correct that the trunk product and the exported carbon should be moved to the first condition. In addition, Michael (Keller) suggested some numbers to use to apply a discount to the actually exported carbon and add the leftover to the large CWD pool ranging from 1% to 3.8% for RIL and CL respectively as a result of limitation in length of transportation vehicles and/or poor planning (trees could be felled but not found). So feel free to let me know if you want to submit a pull request on this, or I can submit it as part of an new issue.

**Comment by rgknox** [comment]
@huangmy , if you would like to submit a PR, that would be great.  Otherwise, I will include this in my nutrient refactors, but the time-table on that is not short term.

**Comment by huangmy** [comment]
@rgknox sure, I will include these in a PR then, likely sometime next week. Thanks.

**Comment by rosiealice** [comment]
Just concurring that the litter_in logic is indeed backwards to my mind.  The way that SPITFIRE handle the trunk index (which I had thought would avoid having the hardwire the C==NSC condition) is a bit diffiult/confusion, in that while we have four NSC classes, spitfire has six fuel classes, and so the indexing in SPITFIRE is not the same as for when we are thinking aboutthe NSC pools (as in, the trunk index in SPITFIRE is 5, not 4.

https://github.com/NGEET/fates/blob/64b5affb69b681c923a6575457eda2785f5b5789/main/EDTypesMod.F90#L134

I can't think of a good way out of this right now, other than to maybe re-order the fiel catagories so that we have the NSC's first and then the live and dead grasses as indices 5 and 6. (right now they are 1 and 6). That way, at least the NSC indicies in SPITFIRE would align with those in the logging and litter code.

Just wanted to get that thought out somewhere. If people think this is potentially a better way of organizing things, I could refactor it as such. Do you have any thoughts on this @jkshuman ?

**Comment by rgknox** [comment]
I like this idea @rosiealice

**Comment by huangmy** [comment]
@rgknox @rosiealice : So do I need to worry about this? Sounds like @rosiealice is proposing to change the indices in SPITFIRE, not in the logging module, correct?

**Comment by jkshuman** [comment]
@rosiealice refactoring so that SPITFIRE and the logging and litter are the same makes sense. I can update this, so I can maintain consistency in my current testing.

**Comment by jkshuman** [comment]
@rgknox can I update this in the most current version of master? or do you have a version you are working with? will need to also update the tech note to reflect this change - is the tech note also on github?

**Comment by rgknox** [comment]
@huangmy , I think fixing the trunk product fix can go as a stand alone contribution or in a broader refactor, either should be fine.  I will step back and let you @jkshuman hash it out the work.
@jkshuman , if you would like to tackle this, then I recommend working off of master, I will make sure to integrate your changes into my broader litter refactors.

**Comment by jkshuman** [comment]
@huangmy I will create a branch with the SPITFIRE refactor to switch dead and live grass to pools 5 and 6 based on the current master. Then can I pass it off to you?

**Comment by huangmy** [comment]
@jkshuman Sounds great to me!

**Comment by jkshuman** [comment]
@rgknox  I have committed the SF fuels refactor changes to SF_fuels_refactor branch on my repo. 
I am getting a failure in testing with a string of ENDRUN statements related to negative leaf and root litter stocks (snippet below). I made changes to indexing in SFMainMod, EDTypes, and within the parameter file. Maybe I missed something else...
Working through a few ideas, but if you have thoughts, let me know. 
```
 ERROR: Unknown error submitted to shr_abort_abort.
 ENDRUN:
 ERROR in EDMainMod.F90 at line 648                                             
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                       
 lat lon  -14.0000000000000        280.000000000000     
 ---------------------------------------
   9972.60273972603       9.341668847675320E-006  6.227778613109289E-006
  0.000000000000000E+000  0.000000000000000E+000
 ---
 pft:            1
 dbh:   0.380142644301348     
 lc:   1.636111334279184E-002  dc:   8.653587763975318E-003  stc: 
  3.272222668558369E-002
 fc:   1.636111334279184E-002  rc:   0.000000000000000E+000  sac: 
  1.149137554440363E-004
 N plant:    1983.48496106071     
 pft:           13
 negative CWD_AG -1.147085595921413E-005  -26.0000000000000     
   315.000000000000     
 negative CWD_BG -7.647236546458011E-006  -26.0000000000000     
   315.000000000000     
 negative leaf litter numerical error -1.606290470669965E-005
  -26.0000000000000        315.000000000000      -1.745623350109958E-002
  0.000000000000000E+000  1.745623350109958E-002  2.739726027397260E-003
 negative root litter numerical error -5.276121762846625E-005
 -1.570899512341566E-004  -26.0000000000000        315.000000000000     
 negative leaf litter numerical error -5.691068464029807E-004
 negative CWD_AG -9.993343230126826E-006  -22.0000000000000     
   300.000000000000     
 negative CWD_BG -6.662228158251606E-006  -22.0000000000000     
   300.000000000000     
 negative leaf litter numerical error -1.402150293684728E-005
  -22.0000000000000        300.000000000000      -1.478639462777173E-002
  1.514947341075305E-005  1.480154410118249E-002  5.479452054794521E-003
 negative root litter numerical error -3.791314656416837E-005
 -1.264380919098764E-004  -22.0000000000000        300.000000000000     
 negative leaf litter numerical error -4.971532780822527E-004
  -22.0000000000000        300.000000000000      -0.524016409263623     
  4.005775224101821E-004  0.524416986786033       5.479452054794521E-003
 negative root litter numerical error -9.482335049838305E-004
 -3.321426227860983E-003  -22.0000000000000        300.000000000000     
 dbh:   0.169354284102829     
 lc:   5.719196143944128E-003  dc:   1.945148558973471E-004  stc: 
  6.863035645445477E-003
 fc:   5.719196143944128E-003  rc:   0.000000000000000E+000  sac: 
  7.530001741132342E-006
 N plant:    50099.5780207225     
 pft:           13
 dbh:   0.172102624817102     
  -26.0000000000000        315.000000000000      -0.618472323610393     
  0.000000000000000E+000  0.618472323610393       2.739726027397260E-003
 negative root litter numerical error -1.328389750191063E-003
 -4.112060032996454E-003  -26.0000000000000        315.000000000000     
 ERROR: Unknown error submitted to shr_abort_abort.
 lc:   5.839675701120504E-003  dc:   2.017479633239963E-004  stc: 
  7.007611119802040E-003
 fc:   5.839675701120503E-003  rc:   0.000000000000000E+000  sac: 
  7.765979471624073E-006
 N plant:    148107.015553635     
 ---------------------------------------
   27.3972602739726       9.315743750039720E-006  6.210495216402508E-006
  0.000000000000000E+000  0.000000000000000E+000
 ---
 pft:           13
 dbh:   0.172137013769164     
 lc:   5.841183216435004E-003  dc:   2.018384684725265E-004  stc: 
 negative CWD_AG -1.188444335751050E-005  -26.0000000000000     
   315.000000000000     
 negative CWD_BG -7.922961451264784E-006  -26.0000000000000     
   315.000000000000     
 negative leaf litter numerical error -1.667414987635330E-005
  -26.0000000000000        315.000000000000      -1.767933798802317E-002
  1.512436281418950E-005  1.769446235083736E-002  5.479452054794521E-003
 negative root litter numerical error -4.583566699345761E-005
 carbon balance error detected
 error fraction relative to biomass stock:  3.952914254501555E-003
 call index:            1
 flux in (npp):     47.2657153686151     
 flux out (fragmentation/harvest):    52.9403784243027     
 net:   -5.67466305568761     
 dstock:    9.46979452352343     
 error=net_flux-dstock:   15.1444575792110     
 biomass   3827.00814283924     
 litter  0.155704215570054     
  7.009420138251326E-003
 fc:   5.841183216435004E-003  rc:   0.000000000000000E+000  sac: 
  7.768932172073300E-006
 N plant:    139.977874672521     
 aborting on date:           1           1           2
 ENDRUN:
 ERROR in EDMainMod.F90 at line 648                                             
```

**Comment by jkshuman** [comment]
Similar error with testing using 2 tree PFTs. @rgknox indicated he is working on a collection of repairs related to these errors as brought up in other open issues. 

``` 
carbon balance error detected
 error fraction relative to biomass stock:  8.675811464041288E-005
 call index:            1
 flux in (npp):    0.549353334090488     
 flux out (fragmentation/harvest):   0.556528312610669     
 net:  -7.174978520181230E-003
 dstock:   1.781188473688644E-002
 error=net_flux-dstock:  2.498686325706767E-002
 biomass   287.867526809622     
 litter  0.120461141441900     
 seeds  1.807765595141700E-002
 previous total   287.988253722278     
 lat lon  -2.00000000000000        310.000000000000     
 ---------------------------------------
   10000.0000000000       5.603940711849675E-006  3.735960103432136E-006
  0.000000000000000E+000  2.706213328908199E-006
 ---
 pft:            1
 dbh:   0.374515919269886     
 lc:   1.604771978076038E-002  dc:   8.363956666512953E-003  stc: 
  3.209543956152076E-002
 fc:   1.604771978076038E-002  rc:   0.000000000000000E+000  sac: 
  1.404534118974673E-005
 N plant:    1983.40888218314     
 pft:            2
 dbh:   0.374515919269886     
 lc:   1.604771978076038E-002  dc:   8.363956666512953E-003  stc: 
  3.209543956152076E-002
 fc:   1.604771978076038E-002  rc:   0.000000000000000E+000  sac: 
  1.404534118974673E-005
 N plant:    1983.40888218314     
 aborting on date:           1           1           3
 ENDRUN:
 ERROR in EDMainMod.F90 at line 648       
  ```

**Comment by jkshuman** [comment]
@huangmy I have this functioning now. Just testing it with and without fire, and will pass it off to you.

**Comment by jkshuman** [comment]
@huangmy The SPITFIRE refactor is complete and in my repo on branch "SF_fuels_refactor" https://github.com/jkshuman/fates/tree/SF_fuels_refactor
I tested it with the 2 Trop Tree default file with and without fire disturbance, and it ran pan-tropically successfully out to 16 years. Let me know if you need any other details, or have any questions.

**Comment by huangmy** [comment]
@rgknox : I just got a chance to look into this. Can you point me to the right CLM tag for testing? Thanks.

**Comment by rgknox** [comment]
I recommend merging with master, and then if you want to compare against an existing tag, I just released: 
sci.1.26.2_api.7.2.0
https://github.com/NGEET/fates/releases/tag/sci.1.26.2_api.7.2.0
We'd also be happy to help with any testing, just let us know.

**Comment by huangmy** [comment]
@rgknox : has Jackie's branch been merged into master yet? Or shall I add my changes to her branch and then merge it?

**Comment by rgknox** [comment]
@huangmy, her branch has not been merged in yet.

**Comment by huangmy** [comment]
@rgknox: I created a fork from the current FATES master and a pull request from Jackie's branch at https://github.com/huangmy/fates/pull/1. Could you please take a look and let me know if it is OK to merge? If there are better ways to do this, please feel free to let me know. Thanks.

**Comment by huangmy** [comment]
@rgknox @jkshuman I have modified the logging codes on top of @jkshuman 's. Please check at https://github.com/huangmy/fates/pull/2 and let me know your comments.

**Comment by ckoven** [comment]
I believe this was fixed via #523, so closing.
